Client/Frontend/Browser/Playlist/Cache/PlaylistManager.swift

```diff
@@ -101,13 +101,37 @@ class PlaylistManager: NSObject {
     }
     
     func sizeOfDownloadedItem(for pageSrc: String) -> String? {
-        if let assetUrl = downloadManager.localAsset(for: pageSrc)?.url,
-           FileManager.default.fileExists(atPath: assetUrl.path),
-           let size = try? FileManager.default.attributesOfItem(atPath: assetUrl.path)[.size] as? Int {
-            let formatter = ByteCountFormatter()
-            formatter.zeroPadsFractionDigits = true
-            formatter.countStyle = .file
-            return formatter.string(fromByteCount: Int64(size))
+        var isDirectory: ObjCBool = false
+        if let asset = downloadManager.localAsset(for: pageSrc),
+           FileManager.default.fileExists(atPath: asset.url.path, isDirectory: &isDirectory) {
+            
+            if isDirectory.boolValue || asset.url.pathExtension.lowercased() == "movpkg" {
+                let properties: [URLResourceKey] = [.isRegularFileKey, .totalFileAllocatedSizeKey]
+                guard let enumerator = FileManager.default.enumerator(at: asset.url,
+                                                                      includingPropertiesForKeys: properties,
+                                                                      options: .skipsHiddenFiles,
+                                                                      errorHandler: nil) else {
+                    return nil
+                }
+                
+                let resources = enumerator.compactMap({ try? ($0 as? URL)?.resourceValues(forKeys: Set(properties)) })
+                                          .filter({ $0.isRegularFile == true })
+                let sizes = resources.compactMap({ $0.totalFileAllocatedSize })
```

Comment:
Since we do this functional convention we can assign it to a single property after all modifications are done, no need to have `resources` constant imo

Reply:
![image](https://user-images.githubusercontent.com/1530031/115450726-4a8b6280-a1ea-11eb-9dca-2cb64a161645.png)

Reply:
I fixed it up by removing the `.reduce` and just putting it in the subsequent expression :D
Seems to work fine.

Reply:
Too early for functional Swift I guess :)